### DISCUSSION TITLE
Room Presence instead of device location

### DISCUSSION THREAD

**Discussion by grayfrost90** [ESPresense/ESPresense#364] [Q&A] [answered]
I've been playing around with ESPresense for the last week and realise that what I'm really after are sensors in HA for each room, rather than sensors for each device, that then tell me if the room is occupied - to keep lights on etc.

Is that possible or do you just need to do that via templates, helpers, automations etc? I was hoping to use our Garmin watches but they don't seem to advertise the ble signal. I've tried using the HA App and the exp:20 from the NHS exposure app - which actually works really well.... until you have more than one phone and then it jumps around the rooms. If I could just set up sensors for the rooms instead and see if there was someone in a room, that would do what I'm after.

thanks

Jack

**Comment by DTTerastar** [comment, marked as answer]
A feature i'm considering is a configuration setting something like query or include that counts the number of things found with that prefix and the esp publishes that as a sensor value.  That way you could add exp:20 and it would count the number of macs w/ that id.

**Reply by grayfrost90** [reply]
Yeah, that sort of thing. For now I've managed to get the HA companion app working and added some template sensors for the rooms

**Reply by markmghali** [reply]
grayfrost90,

Could you share the config for the template sensors for the rooms?

**Reply by grayfrost90** [reply]
sure, no idea if it's a good example or not as I'm still quite new to this :) 
 
```
template:
  - binary_sensor:
      - name: "Dining Room Occupancy"
        device_class: occupancy
        state: >
          {% if is_state('sensor.mqtt_ble_1', 'dining_room') or is_state('sensor.mqtt_ble_2', 'dining_room') %}
            On
          {% else %}
            Off
          {% endif %}
```

replace "mqtt_ble_1" and "mqtt_ble_2" with the name of your ble sensors and add more sensors for each room. I'm yet to actually do anything more with it, that's next on the list!

**Comment by cromelex** [comment]
Just wanted to say something like this would be very handy for automations.

I know we already have reporting filtering where we can selectively chose what ids to report via mqtt. Having the abilty of reporting the count of these is detected by room would be very handy for automations where you want to automate something when no-one is in the room.

In the meantime I might something like the template sensor exemplified above.

Thanks

**Comment by DTTerastar** [comment]
This is totally implemented already: https://espresense.com/configuration/settings#counting

**Reply by cromelex** [reply]
Thanks, I'll give it a go then.
 I had seen that in the documentation but somehow misinterpreted the functionality to mean something different than what I wanted...
Thanks!

**Reply by asimmetrico** [reply]
> This is totally implemented already: https://espresense.com/configuration/settings#counting

I was looking for the same. Having the count is exactly what is needed. I tried it time ago and is awesome.The problem with that is updating.
When you have to add or remove an BLE object (you change your smartwatch or add a person or ...) you have to update any Espresense device in the house. It would be nice to have a function to update all the Espresense device at a time with the list of allowed BLE.

Or there is a way to do so that I don't know?

Best

Al

**Reply by DTTerastar** [reply]
There is you send a mqtt message with /set on the end to the * room.
